lib/improver/blending/weights.py

```diff
@@ -63,13 +68,13 @@ def normalise_weights(weights):
                 ValueError: any negative weights are found in input.
                 ValueError: sum of weights in the input is 0.
         """
-        if weights.min() < 0.0:
+        if np.any(weights.min(axis=axis) < 0.0):
             msg = ('Weights must be positive. The weights have at least one '
                    'value < 0.0: {}'.format(weights))
             raise ValueError(msg)
 
-        sumval = weights.sum()
-        if sumval == 0:
+        sumval = weights.sum(axis=axis, keepdims=True)
```

Comment:
@MoseleyS I found the magic keyword arg keepdims=True keeps the number of dimensions the same as the input, exactly what we needed so I didn't have to use my own line to reshape the sumval for the division to work.

Reply:
:-)